### PR TITLE
add script to manually sync data from individual facilities + cleanup

### PR DESCRIPTION
**Pull request by czue**
can be read commit by commit

@kkrampa fyi

### CODE REVIEW

File: custom/ilsgateway/management/commands/test_data_migration.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from django.core.management import BaseCommand
+from casexml.apps.stock.models import StockTransaction
+from corehq.apps.commtrack.models import StockState
+from custom.ilsgateway.api import ILSGatewayEndpoint
+from custom.ilsgateway.models import ILSGatewayConfig
+from custom.ilsgateway.tasks import get_ilsgateway_data_migrations
+from custom.ilsgateway.utils import get_supply_point_by_external_id
+from custom.logistics.tasks import get_or_create_checkpoint, stock_data_task
```
Comment (dimagimon) [inline custom/ilsgateway/management/commands/test_data_migration.py]: F401 'get_or_create_checkpoint' imported but unused

Comment (dimagimon) [inline custom/ilsgateway/management/commands/test_data_migration.py]: F401 'get_or_create_checkpoint' imported but unused

**Comment by czue** [comment]
:white_check_mark:

File: custom/logistics/tasks.py [hunk 2]
```diff
@@ -66,9 +73,7 @@ def stock_data_task(domain, endpoint, apis, test_facilities=None):
         # todo: see if we can avoid modifying the list of facilities in place
         if idx == 0:
             facilities = facilities_copy
-    save_stock_data_checkpoint(checkpoint, 'product_stock', 100, 0, start_date, None, False)
-    checkpoint.start_date = None
-    checkpoint.save()
+    save_stock_data_checkpoint(checkpoint, 'product_stock', 100, 0, None, None)
```
Comment (kkrampa) [inline custom/logistics/tasks.py]: @czue Date is sent to API in order to determine what data was migrated and from what date we should start next time. Setting this to None at the end of migration will cause fetching all records from API during next migration which is obviously wrong.

File: custom/logistics/tasks.py [hunk 3]
```diff
@@ -9,32 +9,27 @@
...
     try:
         checkpoint = StockDataCheckpoint.objects.get(domain=domain)
-        api = checkpoint.api
-        date = checkpoint.date
-        limit = checkpoint.limit
-        offset = checkpoint.offset
-        location = checkpoint.location
         if not checkpoint.start_date:
-            checkpoint.start_date = start_date
+            checkpoint.start_date = datetime.today()
```
Comment (kkrampa) [inline custom/logistics/tasks.py]: checkpoint.start_date means date when pending migration was started. checkpoint.date is basically date of last successful migration (probably it's worth to change name of this field).

**Comment by kkrampa** [comment]
I'm not sure how you wanted to solve it but it looks like migration doesn't work properly after this changes.

File: custom/logistics/tasks.py (hunk 2 above)
Comment (czue) [inline custom/logistics/tasks.py]: sorry about that. https://github.com/dimagi/commcare-hq/pull/5919

got confused between `date` and `start_date` in my cleanup

**Comment by czue** [comment]
@kkrampa besides the fix in https://github.com/dimagi/commcare-hq/pull/5919 is there anything else wrong with this?

**Pull request merged**